src/Query/TaskGroup.ts

```diff
@@ -30,10 +36,14 @@ export class TaskGroup {
      * It only contains the minimal set of headings required to separate
      * this group of tasks from the previous group of tasks.
      *
+     * If this group of tasks has the same first-level heading as the previous
+     * group of tasks, there is no need to display that first-level heading
+     * to separate these tasks from the previous group.
```

Comment:
this is extremely helpful to understand! thanks! Maybe with this comment the lines 33-34 are not needed anymore ("minimal" being explained much better here).